Source/Artifacts/IArtifacts.cs

```diff
@@ -6,86 +6,59 @@
 namespace Dolittle.SDK.Artifacts
 {
     /// <summary>
-    /// Defines a system for working with <see cref="Artifact"/>.
+    /// Defines a system that knows about <see cref="Artifact"/>.
     /// </summary>
-    public interface IArtifacts
+    /// <typeparam name="TArtifact">The <see cref="Type" /> of the <see cref="Artifact" />.</typeparam>
+    public interface IArtifacts<TArtifact>
+        where TArtifact : Artifact
     {
         /// <summary>
         /// Check if there is a type associated with an <see cref="Artifact" />.
         /// </summary>
         /// <typeparam name="T">CLR type of the artifact.</typeparam>
-        /// <returns><see cref="Artifact"/>.</returns>
-        Artifact HasFor<T>();
+        /// <returns><see cref="bool"/>.</returns>
+        bool HasFor<T>();
 
         /// <summary>
         /// Check if there is a type associated with an <see cref="Artifact" />.
         /// </summary>
         /// <param name="type">CLR type of the artifact.</param>
-        /// <returns><see cref="Artifact"/>.</returns>
-        Artifact HasFor(Type type);
+        /// <returns><see cref="bool"/>.</returns>
+        bool HasFor(Type type);
 
         /// <summary>
         /// Get an <see cref="Artifact"/> from a given type.
         /// </summary>
         /// <typeparam name="T">CLR type of the artifact.</typeparam>
         /// <returns><see cref="Artifact"/>.</returns>
-        Artifact GetFor<T>();
+        TArtifact GetFor<T>();
 
         /// <summary>
         /// Get an <see cref="Artifact"/> from a given type.
         /// </summary>
         /// <param name="type">CLR type of the artifact.</param>
         /// <returns><see cref="Artifact"/>.</returns>
-        Artifact GetFor(Type type);
+        TArtifact GetFor(Type type);
 
         /// <summary>
         /// Get a CLR <see cref="Type"/> for a specific <see cref="Artifact"/>.
         /// </summary>
         /// <param name="artifact"><see cref="Artifact"/> to get for.</param>
```

Comment:
```suggestion
        /// Check if there is an <typeparamref name="TArtifact"/> associated with a <see cref="Type" />.
        /// </summary>
        /// <param name="artifact">The <typeparamref name="TArtifact"/>.</param>
```

All of the other references to `TArtifact` in the comments should also use `<typeparamref name="TArtifact"/>`